pkg/clients/cloud_client.go

```diff
@@ -115,6 +116,39 @@ func (c *CloudAPIClient) GetRegionDetails(ctx context.Context, provider CloudPro
 	return &region, nil
 }
 
+// EnableRegion sends a PATCH request to enable a region and polls until the region is enabled or a timeout is reached.
+func (c *CloudAPIClient) EnableRegion(ctx context.Context, provider CloudProvider) (*CloudRegion, error) {
+	endpoint := fmt.Sprintf("%s/api/region", provider.Url)
+	emptyJSONPayload := bytes.NewBuffer([]byte("{}"))
+	req, err := http.NewRequestWithContext(ctx, http.MethodPatch, endpoint, emptyJSONPayload)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request to enable region: %v", err)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := c.FronteggClient.HTTPClient.Do(req)
```

Comment:
Why does this use the frontegg client? This is the cloud region api, right?

Reply:
Hmm, I think that this has flowed from the initial implementation during the large refactor a few months ago.

As far as I can tell, the main idea seems to have been to reuse the FronteggClient's HTTPClient which already includes the Authorization token.

I'll have to refactor this. I've created an issue to track this and will submit a PR tomorrow! https://github.com/MaterializeInc/terraform-provider-materialize/issues/543

Thanks for pointing it out!